snuba/settings.py

```diff
@@ -5,23 +5,27 @@ def _load_settings(obj=locals()):
     main Snuba directory, or they can provide a full absolute path such as
     `/foo/bar/my_settings.py`."""
 
+    import importlib
+    import importlib.util
     import os
-    import imp
-
-    path = os.path.dirname(__file__)
 
     settings = os.environ.get('SNUBA_SETTINGS', 'base')
     if not settings.startswith('/') and not settings.startswith('settings_'):
-        settings = 'settings_%s' % settings
-    if not settings.endswith('.py'):
+        settings = '.settings_%s' % settings
+    elif not settings.endswith('.py'):
```

Comment:
If a shortname starts with `settings_` underscore here then we'll incorrectly end up adding `.py` to the end. This will fail when we call `import_module` later.

Reply:
I think this was already the case. I'll try to see if I can make this part of the code clearer and stricter.

Reply:
It was the case before, but we always expected the file to have `.py` in the previous implementation. Now we only want to end with `.py` when we're actually opening a file, as opposed to a module path.